sstable/reader_iter_single_lvl.go

```diff
@@ -320,6 +320,24 @@ func disableBoundsOpt(bound []byte, ptr uintptr) bool {
 // state and require this behavior to be deterministic.
 var ensureBoundsOptDeterminism bool
 
+// SetBoundsMaterialized indicates whether this iterator, if wrapped by a prefix
+// rewriting iterator and virtual state, requires keys passed to its SetBounds()
+// method to *not* be rewritten into the "backing" key space, that would match
+// the actual iterator contents. If it returns true, this iterator instead
+// expects bounds to still be in terms of of "materialized" prefix.
+//
+// This allows an optimization in vState, whereby when this iterator passes it
+// bounds to constrain, since bounds are kept in terms of materialized keys, the
+// vState does not need to convert them from the "backing" prefix before it can
+// combine them with the virtual bounds, which are always in terms of
+// materialized keys. The vState then converts its constrained bounds to backing
+// bounds which are what SetBounds will ultimately apply, so essentially this is
+// just indicating that constrainBounds is responsible for the rewriting, rather
+// than the caller of SetBounds (the wrapping PrefixRewritingIterator).
+func (i singleLevelIterator) SetBoundsMaterialized() bool {
```

Comment:
I guess this gets copied through to `twoLevelIterator` seeing as that struct extends/embeds `singleLevelIterator`, but yea this could be enough reason to update the test to pass in tiny block sizes in a second variant to force the use of two level iterators..

Reply:
twoLevelIterator doesn't call `constrainBounds` directly; it _only_ calls it via the embed of SLI and its `SetBounds` call so I'm fine with this.